### PR TITLE
Update to design-system 0.31.0 | Convert `a-heading` inside `m-slug-header` to `m-slug-header_heading`

### PR DESCRIPTION
**Pull request by anselmbradford**
Merge after https://github.com/cfpb/design-system/pull/1732

Contains:
https://github.com/cfpb/design-system/releases/tag/v0.31.0
https://github.com/cfpb/design-system/compare/v0.30.0...v0.31.0

## Changes

- Convert `a-heading` inside `m-slug-header` to `m-slug-header_heading`
- Update meta header to use the one defined in the design system (updated in https://github.com/cfpb/design-system/pull/1731)

## How to test this PR

1. http://localhost:8000/blog post previews meta header (bit above the post title link) is unchanged from production.
1. http://localhost:8000/rural-or-underserved-tool/ (e.g.) will have slug headers in the right-hand side bar that are unchanged from production.

### CODE REVIEW

File: cfgov/v1/jinja2/v1/includes/organisms/post-preview.html [hunk 1]
```diff
@@ -74,7 +74,7 @@
                     {% if loop.index > 1 %}
                         <span class="m-meta-header_separator">|</span>
                     {% endif %}
-                    <span class="m-meta-header_category">{{ category_slug.render(category=cat.name, href=url) }}</span>
+                    <span class="m-meta-header_category">{{ category_slug.render(category=cat.name) }}</span>
```
Comment (anselmbradford) [inline cfgov/v1/jinja2/v1/includes/organisms/post-preview.html:77]: @chosak This "meta header" is only found in post previews and it doesn't appear they are ever links. Is there any other cleanup that can happen here?

**Review by natalia-fitzgerald — APPROVED** [review]

File: cfgov/v1/jinja2/v1/includes/organisms/post-preview.html (hunk 1 above)
Comment (chosak) [reply cfgov/v1/jinja2/v1/includes/organisms/post-preview.html:77]: @anselmbradford the `href` parameter is currently used in the PostPreviewSnapshot module, in the template [here](https://github.com/cfpb/consumerfinance.gov/blob/e19568831dab1f29f417cd6545ab522ce0981e37/cfgov/v1/jinja2/v1/includes/organisms/post-preview-snapshot.html#L28).

**Comment by anselmbradford** [comment]
Hold till we work out what we want to do with
https://www.consumerfinance.gov/compliance/amicus/
https://www.consumerfinance.gov/rules-policy/notice-opportunities-comment/

**Comment by natalia-fitzgerald** [comment]
@anselmbradford 
Let's remove the links from those two instances. The _overwhelming_ majority of instances of the meta header component on our website do not include a link. So we can safely assume that the links were added on these _two_ pages due to an inconsistency approach or human error when building the pages in the Wagtail interface. Both pages include a link to go back to the full list of results and filter so, if needed, a user can access the content and filter by the category that they desire.

**Comment by chosak** [comment]
@natalia-fitzgerald that's great; as Ans alludes to [above](https://github.com/cfpb/consumerfinance.gov/pull/7974#discussion_r1340703243) that change will allow for some additional simplification of the post preview template. @anselmbradford can we go ahead with this PR and then I can open a subsequent one to do that simplification? Or is there something about these changes that requires including that link removal here?

**Comment by anselmbradford** [comment]
> can we go ahead with this PR and then I can open a subsequent one to do that simplification? Or is there something about these changes that requires including that link removal here?

Chatted and we'll go ahead with this PR and then in a subsequent PR clean up the `url` argument that as of this PR is getting ignored in the post preview.

**Pull request merged**